topic05-git-advanced/book/04.04.md

```diff
@@ -0,0 +1,48 @@
+# Tagging
+
+* There are two types of git tags - annotated and lightweight
+* We’re going to only cover lightweight tags today (you’ll see these more often)
+* A lightweight tag is like a branch that does not change
+  * Pointer to a specific git-ref.
+  * Unlike annotated tags, it is not a full object in the Git DB (not annotated, not-checksummed, not signed, no metadata about the creator)
+* Lightweight tags - under the covers
+
+```bash
+# LW tags are files on disk with a pointer to a commit
+ls .git/refs/tags/
+
+# Cat to reveal pointer to a commit
+cat git/refs/tags/v99.0.0
+
+# Show this commit
+git show 659220 
+```
+
+Creating a tag
+
+```bash
+# On a branch or any kind of git-ref
+git tag v1.2.3-your-name
+
+# Check filesystem (ls .git/refs/tags)
+# Git does not push tags by default to a remote
+# Or `git push --tags`
+git push origin v1.2.3-your-name
+
+# Checkout a tag
+git checkout v99.0.0
+
+# Create a branch from a tag
+git checkout -b version-nine-nine v99.0.0
+
+# Delete tags /!\ Careful now  /!\
+# Local only
+git tag -d v99.0.0
+
+# Delete from remote
+git push origin :refs/tags/v99.0.0
+
+# A little bit about annotated tags
+git tag -s -a v0.99.0-annotated -m "my tag v0.99.0-annotated"
```

Comment:
I don't think you need `-s` *and* `-a`. `-s` implies `-a` (but not the other way around)